crates/slumber_core/src/template/render.rs

```diff
@@ -752,26 +752,56 @@ impl<'a> ChainTemplateSource<'a> {
             self.chain_id.to_string()
         };
 
-        let options = future::try_join_all(options.iter().enumerate().map(
-            |(i, template)| {
-                // Fork the local state, one copy for each new branch
-                let mut stack = stack.clone();
-                async move {
-                    template
-                        .render_chain_config(
-                            format!("options[{i}]"),
-                            context,
-                            &mut stack,
-                        )
-                        .await
-                }
-            },
-        ))
-        .await?;
+        let resolved_options: Vec<String> = match options {
+            SelectOptions::Fixed(options) => {
+                future::try_join_all(options.iter().enumerate().map(
+                    |(i, template)| {
+                        // Fork the local state, one copy for each new branch
+                        let mut stack = stack.clone();
+                        async move {
+                            template
+                                .render_chain_config(
+                                    format!("options[{i}]"),
+                                    context,
+                                    &mut stack,
+                                )
+                                .await
+                        }
+                    },
+                ))
+                .await?
+            }
+            SelectOptions::Dynamic(options) => {
+                let dynamic_options = options
+                    .render_chain_config("options", context, stack)
+                    .await?;
+
+                let raw_results: Vec<serde_json::Value> = serde_json::from_str(
+                    &dynamic_options,
+                )
+                .map_err(|error| ChainError::DynamicSelectOptions {
+                    error: Arc::new(error),
+                })?;
+
+                raw_results
+                    .iter()
+                    .map(|value| {
+                        // if the value is already a string, avoid calling
+                        // to_string on it since that would wrap the string
+                        // in another set of quotes.
+                        if value.is_string() {
+                            value.as_str().unwrap().into()
+                        } else {
+                            value.to_string()
+                        }
```

Comment:
Pattern matching is your friend :)

```suggestion
                        match value {
                            serde_json::Value::String(s) => s,
                            other => other.to_string()
                        }
```

Reply:
lol this is my long history with JS/TS coming through 🤦

Reply:
I too am a recovering JS dev. There is hope for you!